be/src/exec/pipeline/hashjoin/spillable_hash_join_build_operator.cpp

```diff
@@ -70,10 +70,24 @@ bool SpillableHashJoinBuildOperator::need_input() const {
 Status SpillableHashJoinBuildOperator::set_finishing(RuntimeState* state) {
     auto defer_set_finishing = DeferOp([this]() { _join_builder->spill_channel()->set_finishing(); });
 
-    if (!_join_builder->spiller()->spilled()) {
+    if (spill_strategy() == spill::SpillStrategy::NO_SPILL ||
+        (!_join_builder->spiller()->spilled() &&
+         _join_builder->hash_join_builder()->hash_table().get_row_count() == 0)) {
         return HashJoinBuildOperator::set_finishing(state);
     }
 
+    // if this operator is changed to spill mode just before set_finishing,
+    // we should create spill task
```

Comment:
CHECK FORCE_SPILLED  before actually to spill?

Reply:
> CHECK FORCE_SPILLED before actually to spill?

add a DCHECK here